src/main/java/com/example/bookstoreapp/controller/CategoryController.java

```diff
@@ -0,0 +1,106 @@
+package com.example.bookstoreapp.controller;
+
+import com.example.bookstoreapp.dto.book.BookDtoWithoutCategoryIds;
+import com.example.bookstoreapp.dto.category.CategoryRequestDto;
+import com.example.bookstoreapp.dto.category.CategoryResponseDto;
+import com.example.bookstoreapp.service.BookService;
+import com.example.bookstoreapp.service.CategoryService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.Positive;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.web.PageableDefault;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Category management ", description = "Endpoints for managing categories")
+@RestController
+@RequestMapping("/categories")
+@RequiredArgsConstructor
+public class CategoryController {
+
+    private final CategoryService categoryService;
+    private final BookService bookService;
+
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
```

Comment:
```suggestion
    @PreAuthorize("hasRole('ADMIN')")
```
you can simplify  it, apply to other endpoints